wire/mixvec.go

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2023 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package wire
+
+import (
+	"fmt"
+	"strings"
+)
+
+// MixVec is a N-element vector of Msize []byte messages.
+type MixVec struct {
+	N     uint32
```

Comment:
the two-dimensional array is not as easy to guarantee it is the proper length, as you must iterate every element and ensure it is the expected size.  With the single backing slice, you only need to verify it is length N * Msize.

I think there's also a small historical reason for this, where the single slice was a little more efficient to encode with gob (used by the client-server mixing code).

If we want to make this code clearer and less prone to error, do you think we should hardcode the Msize (message size)? Right now it is always 20 bytes, because what we are mixing are the pubkey hash (or in the future, perhaps the script hash) of a p2pkh/p2sh output.

Reply:
the other reason I can recall now is that when xoring these vectors together, the single backing slice allowed that xor operation to be implemented in a single loop.  we can still do the slice of arrays, but would need to have inner and outer loops.

I'm thinking we should probably just go with a hardcoded msize=20, check that the slice length is a multiple of 20, and infer the number of elements from the total length.